edx_exams/apps/api/v1/views.py

```diff
@@ -179,3 +180,53 @@ def patch(self, request, course_id):
             data = {"detail": "Invalid data", "errors": serializer.errors}
 
         return Response(status=response_status, data=data)
+
+
+class CourseExamConfigurationsView(APIView):
+    """
+    View to create and update course exam configs for a specific course.
+
+    Given a course id and a proctoring provider name, this view will either create a new course exam configuration
+    (if one doesn't exist), or modify the proctoring provider on an existing course exam config.
+
+    HTTP PATCH
+    Creates or updates a CourseExamConfiguration.
+    Expected PATCH data: {
+        'provider': 'test_provider',
+    }
+    **PATCH data Parameters**
+        * name: This is the name of the proctoring provider.
+    **Exceptions**
+        * HTTP_400_BAD_REQUEST
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserPermissions,)
+
+    @classmethod
+    def handle_config(cls, provider, course_id):
+        """
+        Helper method that decides whether to update existing or create new config.
+        """
+        CourseExamConfiguration.objects.update_or_create(
+            course_id=course_id,
+            defaults={'provider': provider})
+        log.info(f"Created or updated course exam configuration course_id={course_id},provider={provider.name}")
+
+    def patch(self, request, course_id):
+        """
+        Create/update course exam configuration.
+        """
+        try:
+            provider = ProctoringProvider.objects.get(name=request.data['provider'])
```

Comment:
The `ProctoringProvider` doesn't have a uniqueness constraint on `name`, so it's possible for this to return more than one provider. We should either add a uniqueness constraint on the field or handle the case where more than one provider is returned.

Reply:
That's a good point! I can create another PR as I did for adding the uniqueness constraint to the course exam config.

Reply:
Here's [that PR](https://github.com/edx/edx-exams/pull/30)!

Reply:
Merged the above!